src/Sentry/SentryClient.cs

```diff
@@ -102,27 +101,27 @@ public SentryId CaptureEvent(SentryEvent @event, Scope scope = null)
         /// <returns>A task to await for the flush operation.</returns>
         public Task FlushAsync(TimeSpan timeout) => Worker.FlushAsync(timeout);
 
-        private SentryId DoSendEvent(SentryEvent @event, Scope scope)
+        // TODO: this method needs to be refactored, it's really hard to analyze nullability
+        private SentryId DoSendEvent(SentryEvent @event, Scope? scope)
         {
-            if (_options.SampleRate is float sample)
+            if (_options.SampleRate != null)
             {
-                if (Random.NextDouble() > sample)
+                if (Random.NextDouble() > _options.SampleRate.Value)
                 {
                     _options.DiagnosticLogger?.LogDebug("Event sampled.");
                     return SentryId.Empty;
                 }
             }
-            if (@event.Exception is Exception ex
-                && _options.ExceptionFilters is IExceptionFilter[] filters && filters.Length > 0)
+            if (@event.Exception != null && _options.ExceptionFilters?.Length > 0)
             {
-                if (filters.Any(f => f.Filter(ex)))
+                if (_options.ExceptionFilters.Any(f => f.Filter(@event.Exception)))
                 {
                     _options.DiagnosticLogger?.LogInfo(
-                        "Event with exception of type '{0}' was dropped by an exception filter.", ex.GetType());
+                        "Event with exception of type '{0}' was dropped by an exception filter.", @event.Exception.GetType());
```

Comment:
This doesn't warn that `@event.Exception` is nullable? The code before the change made sure it wasn't through pattern matching. Which IMHO is a better approach to deal with this

Reply:
I see the check above